tasks/task_2/js/phoneBook.js

```diff
@@ -1,52 +1,183 @@
 'use strict';
 
-var phoneBook; // Здесь вы храните записи как хотите
+var fs = require('fs');
+var phoneBook = [];
 
-/*
-   Функция добавления записи в телефонную книгу.
-   На вход может прийти что угодно, будьте осторожны.
-*/
-module.exports.add = function add(name, phone, email) {
+/**
+ * Creates new Contact.
+ * @class
+ * @param {String} name
+ * @param {String} phone
+ * @param {String} email
+ */
+function Contact(name, phone, email) {
+    this.name = name;
+    this.phone = phone;
+    this.email = email;
+}
 
-    // Ваша невероятная магия здесь
+/**
+ * Validates a string using regExp.
+ * @class
+ * @param {String} str
+ * @param {Object} reg
+ */
+function validateStr(str, reg) {
+    return reg.test(str);
+}
 
-};
+/**
+ * Checks if string matches query or not.
+ * @param {String} str
+ * @param {String} query
+ * @returns {Boolean}
+ */
+function isMatch(str, query) {
+    return str.toLowerCase().indexOf(query.toLowerCase()) !== -1;
+}
 
-/*
-   Функция поиска записи в телефонную книгу.
-   Поиск ведется по всем полям.
-*/
-module.exports.find = function find(query) {
+/**
+ * Finds or removes Contacts according to query.
+ * @param {Array} arr
+ * @param {String} query
+ * @param {Boolean} remove
+ * @returns {Array}
+ */
+function findOrRemove(arr, query, remove) {
+    if (remove) {
+        return arr.filter(function (contact) {
+            return !isMatch(contact.name, query) &&
+                !isMatch(contact.phone, query) &&
+                !isMatch(contact.email, query); // returns contacts which don't match query (so we can remove others)
+        });
+    } else {
+        return arr.filter(function (contact) {
+            return isMatch(contact.name, query) ||
+                isMatch(contact.phone, query) ||
+                isMatch(contact.email, query); // returns contacts, in which one+ field matches query
+        });
+    }
+}
 
-    // Ваша удивительная магия здесь
+/**
+ * Looks for repeats in phoneBook.
+ * @param {string} name
+ * @returns {Boolean}
+ */
+function findRepeat(str) {
+    var counter = 0;
+    phoneBook.forEach(function (contact) {
```

Comment:
``for (var i = 0; i < phoneBook.length; i++) { 
    if(phoneBook[i].name === str) return true;
}
return false;```